.github/workflows/build-publish-documentation.yaml

```diff
@@ -5,11 +5,18 @@ on:
   push:
     branches:
       - main
+  pull_request:
+    branches:
+      - main
 
 jobs:
   github-pages:
     runs-on: ubuntu-latest
 
+    strategy:
+      matrix:
+        python-version: [ "3.10" ]
```

Comment:
Is there any reason to use a `matrix` here when we only will ever do this using one Python version?

Reply:
Good point. I saw this pattern in workflows in other repos like linkml – https://github.com/linkml/linkml/blob/main/.github/workflows/doc-pages.yaml#L10-L11 and I thought that we should change it, but left it this way to be consistent with other repos.

But yes you're absolutely right, we're going to be building and publishing the docs using only one version of Python and there's no reason to have a python-version matrix like we have above for that.

I'll go ahead and make the change to correct that. Will make PR's to fix this in other repos as well.